nselib/http.lua

```diff
@@ -464,84 +494,73 @@ local function recv_header(s, partial)
 end
 
 -- Receive until the connection is closed.
-local function recv_all(s, partial)
-  local parts
-
-  partial = partial or ""
-
-  parts = {partial}
-  while true do
-    local status, part = s:receive()
-    if not status then
-      break
-    else
-      parts[#parts + 1] = part
+local function recv_all(s, partial, maxlen)
+  local parts = {}
+  local part = partial or ""
+  repeat
+    if maxlen then
+      maxlen = maxlen - #part
+      if maxlen < 0 then
+        table.insert(parts, part:sub(1, maxlen - 1))
+        return nil, ERR_OVERSIZED_BODY, table.concat(parts)
```

Comment:
In a future revision, we might consider simply receiving and discarding the rest of the body in this case, at least while pipelining. It would take more time and bandwidth than just dropping the connection, but it would avoid missing later response bodies in the pipeline. I'm not going to suggest you hold up this PR just to make this change, but consider it for future improvement.

Reply:
At the moment I am leaning against this alternative. In my opinion we should prevent getting stuck with potentially infinite responses.

If somebody wants to take advantage of the persistent connection but to avoid resubmitted requests then it is always possible to set `http.max-pipeline=1`.